data-upgrade-users/src/main/java/org/exoplatform/migration/UsersLastLoginTimeMigration.java

```diff
@@ -0,0 +1,115 @@
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.exoplatform.migration;
+
+import org.exoplatform.commons.search.index.IndexingService;
+import org.exoplatform.commons.upgrade.UpgradeProductPlugin;
+import org.exoplatform.commons.utils.CommonsUtils;
+import org.exoplatform.commons.utils.ListAccess;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.organization.OrganizationService;
+import org.exoplatform.services.organization.User;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.model.Profile;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.jpa.search.ProfileIndexingServiceConnector;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.social.core.profile.ProfileFilter;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class UsersLastLoginTimeMigration extends UpgradeProductPlugin {
+  private static final Log    LOG = ExoLogger.getExoLogger(UsersLastLoginTimeMigration.class);
+
+  private OrganizationService organizationService;
+
+  private IdentityManager     identityManager;
+
+  public UsersLastLoginTimeMigration(OrganizationService organizationService,
+                                     IdentityManager identityManager,
+                                     InitParams initParams) {
+    super(initParams);
+    this.organizationService = organizationService;
+    this.identityManager = identityManager;
+  }
+
+  @Override
+  public void processUpgrade(String oldVersion, String newVersion) {
+    LOG.info("Start upgrade process to add lastLoginTime in user profile");
+    long startupTime = System.currentTimeMillis();
+    try {
+      int limit = 100;
+      int offset = 0;
+      int totalSize = 0;
+      int totalItemsChecked = 0;
+      ListAccess<Identity> listIdentity = null;
+      ProfileFilter filter = new ProfileFilter();
+      filter.setConnected(false);
+      listIdentity = identityManager.getIdentitiesByProfileFilter(OrganizationIdentityProvider.NAME, filter, true);
+      totalSize = listIdentity.getSize();
+      LOG.info("Number of users to check : " + totalSize);
+      do {
+        int limitToFetch = limit;
+        List<Identity> identities;
+        if (totalSize < (offset + limitToFetch)) {
+          limitToFetch = totalSize - offset;
+        }
+        identities = Arrays.asList(listIdentity.load(offset, limitToFetch));
+        int numberOfModifiedItems = updateLastLoginTime(totalSize, totalItemsChecked, identities);
+        totalItemsChecked = totalItemsChecked + identities.size();
+        offset = (offset + limitToFetch) - numberOfModifiedItems;
+      } while (offset < totalSize);
+      if (totalSize == totalItemsChecked) {
```

Comment:
if updateLastLoginTime throw an error, you will go directly in the catch, and this if will not be evaluated.

Should we throw and error directly in the catch block ?